### ISSUE TITLE
Duplicate errors with `noEmitOnError`

### ISSUE THREAD

**Issue by demurgos** [ivogabe/gulp-typescript#543]
When the option `notEmitOnError` is set to true, `gulp-typescript` reports all the errors twice. This is specific to `gulp-typescript`, there is no issue when passing this option to `tsc`.

The easiest way to understand it is to see an example (see below for the source code):

**Actual behavior:**

Unique errors are reported twice when using the option `noEmitOnError`.

```bash
$ gulp
[23:47:11] Using gulpfile /data/projects/various/duplicate-ts-gulp/gulpfile.js
[23:47:11] Starting 'default'...
src/index.ts(1,7): error TS2322: Type 'number' is not assignable to type 'string'.
src/index.ts(1,7): error TS2322: Type 'number' is not assignable to type 'string'.
[23:47:12] TypeScript: 1 semantic error
[23:47:12] TypeScript: 1 emit error
[23:47:12] TypeScript: emit failed
[23:47:12] Finished 'default' after 889 ms
```

**Expected behavior:**

Unique errors should only be reported once.

```bash
$ gulp
[23:47:11] Using gulpfile /data/projects/various/duplicate-ts-gulp/gulpfile.js
[23:47:11] Starting 'default'...
src/index.ts(1,7): error TS2322: Type 'number' is not assignable to type 'string'.
[23:47:12] TypeScript: 1 semantic error
[23:47:12] TypeScript: 1 emit error
[23:47:12] TypeScript: emit failed
[23:47:12] Finished 'default' after 889 ms
```

---

You can easily reproduce this error with the three following files (or [just clone this temporary repo](https://github.com/demurgos/duplicate-ts-gulp)):

**package.json**
```json
{
  "name": "duplicate-ts-gulp",
  "version": "1.0.0",
  "private": true,
  "devDependencies": {
    "gulp": "^3.9.1",
    "gulp-typescript": "^3.2.3",
    "typescript": "^2.5.3"
  }
}
```

**gulpfile.js**
```javascript
var gulp = require('gulp');
var ts = require('gulp-typescript');

gulp.task('default', function () {
  return gulp.src('src/**/*.ts')
    .pipe(ts({noEmitOnError: true}))
    .pipe(gulp.dest('build'));
}); 
```

**src/index.ts**
```typescript
const foo: string = 1 + 1;
```

**Comment by demurgos** [comment]
The problem occurs here:
https://github.com/ivogabe/gulp-typescript/blob/6914e2260af59a9ef533df3d6f1da327f8d46efe/lib/compiler.ts#L144-L151

With `{noEmitOnError: false}`, I get the following output for `emitOutput`:
```
{ emitSkipped: false,
  diagnostics: [],
  emittedFiles: undefined,
  sourceMaps: undefined }
```

But with `{noEmitOnError: true}`, I get:
```
{ diagnostics: 
   [ { file: [Object],
       start: 6,
       length: 3,
       code: 2322,
       category: 1,
       messageText: 'Type \'1\' is not assignable to type \'string\'.' } ],
  sourceMaps: undefined,
  emittedFiles: undefined,
  emitSkipped: true }
```

**Comment by demurgos** [comment]
I pushed a test-case demonstrating the issue on [my branch `issue-543`](https://github.com/demurgos/gulp-typescript/tree/issue-543). I added a test directory called `noEmitOnError`: this test is currently failing because of the duplicate emit.

A possible explanation is that the error seems to be counted both as a semantic error and emit error.

**Comment by ivogabe** [comment]
Thanks for reporting and investigating this issue. I think this is an issue of TypeScript, I do not think that we should remove duplicate errors ourselves in gulp-typescript. So I reported Microsoft/TypeScript#20876, and hope that we'll get a response from them soon.

**Comment by demurgos** [comment]
The related TS issue was closed [by this PR](https://github.com/Microsoft/TypeScript/pull/21352).

**Comment by demurgos** [comment]
Thanks for fixing this!
